src/pcapi/core/mails/backends/mailjet.py

```diff
@@ -0,0 +1,105 @@
+from datetime import datetime
+from typing import Iterator
+
+from flask import current_app as app
+from requests import Response
+
+from pcapi import settings
+from pcapi.utils.logger import logger
+
+from ..models import MailResult
+from .base import BaseBackend
+
+
+def _add_template_debugging(message_data: dict) -> None:
+    message_data["TemplateErrorReporting"] = {
+        "Email": settings.DEV_EMAIL_ADDRESS,
+        "Name": "Mailjet Template Errors",
+    }
+
+
+class MailjetBackend(BaseBackend):
+    def _send(self, recipients: Iterator[str], data: dict) -> MailResult:
+        data["To"] = ", ".join(recipients)
+
+        if settings.MAILJET_TEMPLATE_DEBUGGING:
+            messages_data = data.get("Messages")
+            if messages_data:
+                for message_data in messages_data:
+                    _add_template_debugging(message_data)
+            else:
+                _add_template_debugging(data)
+
+        try:
+            response = app.mailjet_client.send.create(data=data)
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.exception("Error trying to send e-mail with Mailjet: %s", exc)
+            return MailResult(
+                sent_data=data,
+                successful=False,
+            )
+
+        successful = response.status_code == 200
+        if not successful:
+            logger.warning("Got %d return code from Mailjet: content=%s", response.status_code, response.content)
+
+        return MailResult(
+            sent_data=data,
+            successful=successful,
+        )
+
+    def create_contact(self, email: str) -> Response:
+        data = {"Email": email}
+        return app.mailjet_client.contact.create(data=data)
+
+    def update_contact(self, email: str, *, birth_date: datetime, department: str) -> Response:
+        birth_timestamp = int(datetime(birth_date.year, birth_date.month, birth_date.day).timestamp())
+
+        data = {
+            "Data": [
+                {"Name": "date_de_naissance", "Value": birth_timestamp},
+                {"Name": "département", "Value": department},
+            ]
+        }
+        return app.mailjet_client.contactdata.update(id=email, data=data)
+
+    def add_contact_to_list(self, email: str, list_id: str) -> Response:
+        data = {
+            "IsUnsubscribed": "false",
+            "ContactAlt": email,
+            "ListID": list_id,
+        }
+        return app.mailjet_client.listrecipient.create(data=data)
+
+
+class OverridenMailjetBackend(MailjetBackend):
```

Comment:
Je trouve le nom assez peu clair. Je n'ai pas de super idée. Peut-être quelque chose comme:
```suggestion
class ToDevMailjetBackend(MailjetBackend):
```

Reply:
ou `SandboxMailjetBackend`

Reply:
Je n'aime pas (du tout) le préfixe sandbox, car pour moi "sandbox" c'est un bac à sable **de données**.

`ToDevMailjetBackend`, éventuellement. `OverridenRecipientsMailjetBackend` pour être plus générique mais plus précis ? @cgaunet : un avis ?

Reply:
Plutôt d'avis du ToDevMailJetBackend si on spécifie pas autre chose dans les recipients.
Je ne vois pas le besoin à date d'envoyer ça ailleurs donc je rejoins Xavier

Reply:
C'EST UN COMPLOT !

Je m'incline, `ToDevMailJetBackend` ce sera. :)